bitrise/setup.go

```diff
@@ -70,22 +70,23 @@ func RunSetup(appVersion string, isFullSetupMode bool, isCleanSetupMode bool) er
 		}
 	}
 
-	if err := doSetupBitriseCoreTools(); err != nil {
+	if err := doSetupBitriseCoreTools(setupMode); err != nil {
 		return fmt.Errorf("Failed to do common/platform independent setup, error: %s", err)
 	}
 
-	switch runtime.GOOS {
-	case "darwin":
-		if err := doSetupOnOSX(isFullSetupMode); err != nil {
-			return fmt.Errorf("Failed to do MacOS specific setup, error: %s", err)
+	if setupMode != SetupModeMinimal {
```

Comment:
`doSetupOnOSX` ensures brew is installed, which is a required tool due to the step dependency support. I don't think we can avoid this setup because of the potential step dependencies.